src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -406,6 +406,16 @@ private async Task<PackageSpec> GetPackageSpecAsync(ISettings settings)
                 }
             }
 
+            string enableAudit = _vsProjectAdapter.BuildProperties.GetPropertyValue(ProjectBuildProperties.NuGetAudit);
+            string auditLevel = _vsProjectAdapter.BuildProperties.GetPropertyValue(ProjectBuildProperties.NuGetAuditLevel);
+            RestoreAuditProperties auditProperties = !string.IsNullOrEmpty(enableAudit) || !string.IsNullOrEmpty(auditLevel)
```

Comment:
I am curious what happens when customers set `auditLevel` and don't set `enableAudit` by mistake. As per the current logic, an instance of `RestoreAuditProperties` type will be created if one of them is set.

Reply:
As stated previously, a `PackageSpec` is an abstracted representation of project inputs. If that's how the project is authored, then it's an accurate representation. It also ties to your other comment, because if at some point we flip the default so that audit is enabled by default, unless explicitly opt-out, then you'd want the `RestoreAuditProperties` to contain the AuditLevel, even hwne `EnableAudit` is null/false.